docs/docs-beta/docs/reference/dagster-yaml.md

```diff
@@ -0,0 +1,335 @@
+# dagster.yaml Reference
+
+The `dagster.yaml` file is used to configure the Dagster instance. It defines various settings for storage, run execution, logging, and other aspects of a Dagster deployment.
+
+## File Location
+
+By default, Dagster looks for the `dagster.yaml` file in the directory specified by the `DAGSTER_HOME` environment variable.
+
+## Using Environment Variables
+
+You can use environment variables to override values in the `dagster.yaml` file.
+
+```yaml
+instance:
+  module: dagster.core.instance
+  class: DagsterInstance
+  config:
+    some_key:
+      env: ME_ENV_VAR
+```
+
+## Full Configuration Specification
+
+Here's a comprehensive `dagster.yaml` specification with all available options:
+
+```yaml
+local_artifact_storage:
+  module: dagster.core.storage.root
+  class: LocalArtifactStorage
+  config:
+    base_dir: "/path/to/dir"
+
+compute_logs:
+  module: dagster.core.storage.local_compute_log_manager
+  class: LocalComputeLogManager
+  config:
+    base_dir: /path/to/compute/logs
+  # Can use S3, GCS, or Azure Blob Storage for compute logs, e.g.
+  # module: dagster_aws.s3.compute_log_manager
+  # class: S3ComputeLogManager
+  # config:
+  #   bucket: "mycorp-dagster-compute-logs"
+  #   prefix: "dagster-test-"
+
+storage:
+  sqlite:
+    base_dir: /path/to/sqlite/storage
+  # Or use postgres:
+  # postgres:
+  #   postgres_db:
+  #     hostname: localhost
+  #     username: dagster
+  #     password: dagster
+  #     db_name: dagster
+  #     port: 5432
```

Comment:
Might be worth using environment variables in this example, as that is what people should end up doing.